api/envoy/extensions/filters/http/thrift_to_metadata/v3/thrift_to_metadata.proto

```diff
@@ -0,0 +1,137 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.thrift_to_metadata.v3;
+
+import "envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto";
+
+import "google/protobuf/struct.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.thrift_to_metadata.v3";
+option java_outer_classname = "ThriftToMetadataProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/thrift_to_metadata/v3;thrift_to_metadatav3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Thrift-To-Metadata Filter]
+//
+// The Thrift to Metadata filter serves for thrift over HTTP traffic, expecting serialized
+// Thrift request and response bodies in the HTTP payload. It extracts thrift metadata from the
+// HTTP body and put them into the filter metadata. This is useful for matching load balancer
+// subsets, logging, etc.
+//
+// Thrift to Metadata :ref:`configuration overview <config_http_filters_thrift_to_metadata>`.
+// [#extension: envoy.filters.http.thrift_to_metadata]
+
+enum Field {
```

Comment:
Should this live in core `netowrk.thrift.v3` with other defs? I.e. are these enum values defined by the Thrift standard or us? @zuercher

Reply:
i can answer the enum part. 

these values are referred from https://github.com/envoyproxy/envoy/blob/main/source%2Fextensions%2Ffilters%2Fnetwork%2Fthrift_proxy%2Fmetadata.h#L51-L57 which is inferred or extract from the thrift message of different transport/protocol we support.

spec doesn't have definition of `metadata`. `metadata` is kinda our derivative or thrift and we gives those information for thrift filters.

For example, this is the place we extract seq_id and method name from binary protocol
https://github.com/apache/thrift/blob/master/doc/specs/thrift-binary-protocol.md#message

header flags lives only in header transport, so will be absl::nullopt for other transport
https://github.com/apache/thrift/blob/master/doc/specs/HeaderFormat.md

thrift message itself doesn't specify it's transport, so we either "know" this before we extract, or we determine by magic prefix bytes.